learn/models/conformal-regression/index.qmd

```diff
@@ -0,0 +1,437 @@
+---
+title: "Conformal inference for regression models"
+categories:
+  - probably
+  - parsnip
+  - tune
+  - regression
+type: learn-subsection
+weight: 5
+description: | 
+  The probably package has functions to create prediction intervals for regression models.
+toc: true
+toc-depth: 2
+include-after-body: ../../../resources.html
+---
+
+```{r}
+#| label: "setup"
+#| include: false
+#| message: false
+#| warning: false
+source(here::here("common.R"))
+```
+
+
+```{r}
+#| label: startup
+#| include: false
+#| cache: false
+library(tidymodels)
+library(probably)
+library(doMC)
+
+# -------------------------------------------------------------------------
+
+tidymodels_prefer()
+theme_set(theme_bw())
+options(pillar.advice = FALSE, pillar.min_title_chars = Inf)
+registerDoMC(cores = parallel::detectCores())
+
+# ------------------------------------------------------------------------------
+
+pkgs <- c("tidymodels", "probably", "nnet")
+```
+
+`r article_req_pkgs(pkgs)` The probably package should be version 1.0.1 or greater.
+
+What is [conformal inference](https://en.wikipedia.org/wiki/Conformal_prediction)? It is a collection of statistical methods that are mostly used to construct prediction intervals (or prediction sets) for any type of regression or classification model. The basic idea revolves around some theory on how to construct probability statements for a population of data points. 
+
+Since this article is focused on regression problems, we'll motivate the methodology in the context of a numeric outcome. For example, suppose we have collected a set of 1,000 standard normal data points. 
+
+```{r}
+#| label: normal-dist
+#| out-width: 70%
+#| fig-width: 6
+#| fig-height: 4
+
+library(tidymodels)
+tidymodels_prefer()
+
+set.seed(1)
+reference_data <- tibble(data = rnorm(1000))
+
+reference_data %>% 
+  ggplot(aes(x = data)) +
+  geom_line(stat = "density")
+```
+
+If we had a new observation that we thought might be from the same distribution, how would we say (probabilistically) whether we believe that it belongs to the original distribution? 
+
+If we thought that 1,000 were a sufficient sample size, we might compute some quantiles of these data to define "the mainstream of the data." Let's use the 5th and 95th quantiles to set boundaries that define what we would expect to see most of the time:   
+
+```{r}
+#| label: normal-quant
+#| out-width: 70%
+#| fig-width: 6
+#| fig-height: 4
+quants <- quantile(reference_data$data, probs = c(0.05, 0.95))
+
+reference_data %>% 
+  ggplot(aes(x = data)) +
+  geom_line(stat = "density") + 
+  geom_vline(xintercept = quants, col = "red", lty = 2)
+```
+
+If we were to get a new sample beyond these data, we would be able to say that we are about 90% sure that the new sample does not conform to the original distribution. This works under the assumption that the data are exchangeable. 
```

Comment:
I'm getting stuck on this sentence.

If we are talking confidence intervals, I think that sentence is skirting between "the interval covers the parameter" and "the parameter lies in the interval". But since we are talking about prediction intervals, does this still apply? 

What I'm trying to say: is this sentence statistically correct? I can't judge that immediately and my training on how to talk about p-values is kicking in - but it might be misplaced here.

Reply:
I rewrote this (emphasis added here to show the differences): 

> If we compute a prediction on a new unknown sample, we could center this reference distribution around its predicted value. For some _significance_ level, we now know the range of _sample_ values that "conform" to the _variance seen in the_ reference distribution. That range can define our prediction interval.

Reply:
The use of "predicted values" instead of "sample values" was incorrect.